src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeParse.cs

```diff
@@ -3984,6 +3984,11 @@ private static bool ParseByFormat(
             DateTimeFormatInfo dtfi,
             scoped ref DateTimeResult result)
         {
+            // Add a flag to track if we've seen day specifiers
+            if (!parseInfo.seenDaySpecifier && format.GetChar() == 'd')
+            {
+                parseInfo.seenDaySpecifier = true;
+            }
```

Comment:
Thanks again for submitting the PR. 

This is not going to work if the `d` is encountered after the `MMM`. for example you can have format like `MMM dd`. 

I would suggest to just pass the `format` to the method `MatchAbbreviatedMonthName` and create a helper method to check if the `format.Value` contains `d`. Then call this method with the condition 

```C#
                if ((dtfi.FormatFlags & DateTimeFormatFlags.UseGenitiveMonth) != 0 && FormatHasdSpecifier(format))
```

Note, you want to escape the quotations when searching for the `d`